### ISSUE TITLE
Out of sync between master (v0.14.7 Linux) and syncthing-android (v0.14.6)

### ISSUE THREAD

**Issue by jpjp** [syncthing/syncthing-android#752]
I have a Linux Syncthing master where files are written to about once a month.
The only other syncthing node for this folder is syncthing-android.
About once a day or two days I get a warning on the master that the folder is out of sync, and the option to override changes.
If I override those changes the files it says are out of sync are replaced with files with the same md5sum.
I have tried turning off the timestamps option on the master.
Is this a bug in syncthing, or is there something I should try next? Thank you.

**Comment by Nutomic** [comment]
Ping @calmh @AudriusButkevicius

**Comment by dumischbaenger** [comment]
I experienced the same behaviour. Syncthing says folder is out of sync. If I press the override button the same message comes up after a while.

**Comment by AudriusButkevicius** [comment]
Are the timestamps the same on the non-master after this?

**Comment by dumischbaenger** [comment]
Ok - at the moment I can see this behavior again. I checked one of my this files.

On my PC:
-rw-r--r-- 1 syncthing syncthing 345630 Oct 12 21:29 2016-09-27_comstern_lg_stylus_2_bernhard.jpg

On my Android phone:
-rwxr-x--- u0_a102  everybody   345630 2016-10-12 21:29 2016-09-27_comstern_lg_stylus_2_bernhard.jpg

Next I will remove the master status of my PC and investigate further.

**Comment by dumischbaenger** [comment]
Now I copied the file with parameter -a to a different location and removed master status from this folder and synced. After the sync I compared with "ls -l --full-time" and here is the result

> -rw-r--r-- 1 syncthing syncthing 345630 2016-10-12 21:29:13.126484589 +0200 2016-09-27_comstern_lg_stylus_2_bernhard.jpg
>-rw-r--r-- 1 syncthing syncthing 345630 2016-10-12 21:29:12.000000000 +0200 Rechnungen/2016-09-27_comstern_lg_stylus_2_bernhard.jpg

**Comment by mittwinter** [comment]
Although, this issue has been quiet for some time now, yet still open, I want to chime in:

I have the same constellation as the original reporter: A Linux desktop with syncthing (v. 0.14.23) and a shared folder configured to "send only" to a syncthing-android instance. The shared folder contains around 65 GiB of data, and repeatedly larger portions of that are marked as out of sync items.

I would be glad, if this could be fixed or worked around. I'm happily willing to provide any further information needed on this. Thanks.

**Comment by AudriusButkevicius** [comment]
There is not much we can do as android does not support chtimes. The workaround is not to mark android as master or not modify files on the desktop.

**Comment by dumischbaenger** [comment]
Would it be possible to add a "Slave" button to the Android client?

If set the client should only accept changes but never send changes.

**Comment by mittwinter** [comment]
Thanks for your quick reply!

I am not sure whether I got this right, but I have _not_ marked the folder in syncthing on Android as Master.
On the desktop, I have configured the shared folder as "Send Only" (which is the newer term for Master, I guess). Modifications originate from the desktop alone (i. e., from time to time, I copy new files into the shared folder).
On the Android side, the folder is _not_ set to Master (in syncthing-android), i. e. it has "Send & Receive" instead of "Send only" in the web interface configuration of the folder (as well as the usual "Ignore permissions" on Android).

So, in summary, as none of the workarounds really apply to me, is there something else going on here?

**Comment by AudriusButkevicius** [comment]
Right, so this will probably not work and the folder will always look out of sync due to chtimes not working on android and potentially running on a fat32 filesystem. Only alternative is not to have a send only folder and let tye timestamp changes propagate back to the desktop.

**Comment by wweich** [comment]
You can sync inside the folder `/storage/emulated/0/Android/data/com.nutomic.syncthingandroid/`. There the mtimes can be set, so everything should be in sync.

There is also a receive only folder type in the works which will help in the future.

**Comment by dumischbaenger** [comment]
> You can sync inside the folder /storage/emulated/0/Android/data/com.nutomic.syncthingandroid/.
> There the mtimes can be set, so everything should be in sync.

I already do that. I guess it has something to do with the FAT time resolution as I mentioned in a earlier post.

> There is also a receive only folder type in the works which will help in the future.

I think this is the solution. 

Thank you for your effort!

**Comment by calmh** [comment]
We're fighting a trench war against Android here and I did think our latest weapon against its mtime insanity would work. But it seems Android has further tricks up its sleeve that we haven't figured out yet. Maybe the good old media manager think trampling all over the mtimes, same way it seems to resurrect deleted files?

**Comment by juan-g** [comment]
Found a workaround. Syncthing for Android was saying "Out of sync" for the synced folder. The only solution was to locate a file that had different timestamp even after sync, and just edit the file with a text editor on my laptop. Sync works again well between phone and laptop after that.

**Comment by juan-g** [comment]
"Out of sync" happened again on my Android phone and my workaround didn't work. What fixed it was deleting a hidden symlink file (on the synced Linux laptop) that had appeared for an unknown reason, probably unrelated to Syncthing.

**Comment by sudoman** [comment]
One workaround is to change local timestamps of files in your Master copy with this script:

````
#! /bin/bash

## touch files so that their mtimes match up with mtimes on fat partitions.
## (useful for syncing)

file="$1"
mtime="$(stat --printf "%Y" "$file")"
new_mtime="$(expr "$mtime" / 2 '*' 2)"
touch --date="@$new_mtime" "$file"
````

Run it like so: `find my_sync_dir -exec even-touch {} \;` However you'll need to run it every time you add files to your master copy.
